src/pages/admin/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react'
+import { doc, DocumentData, getDoc } from '@firebase/firestore'
+
+import { db } from '@/components/Firebase/init'
+import NavBar from '@/components/NavBar'
+import Button from '@/components/UI/button'
+import { useAuth } from '@/context/Firebase/Auth/context'
+import mod from '@/lib/temp/firebase/functions/setRole'
+
+const Admin = () => {
+  const { currentUser, isAdmin, getUserToken } = useAuth()
+  const [userDoc, setUserDoc] = useState<DocumentData>()
+
+  const getUserDoc = useCallback(async () => {
```

Comment:
I realise this is just a POC, but you can replace all of this logic by just calling `useUser` BTW

Reply:
yea but im also retrieving doc from a different collection.